### DISCUSSION TITLE
<Link> Use Help

### DISCUSSION THREAD

**Discussion by besscroft** [vercel/next.js#64164] [Help]
### Summary

Hello the community,

The `<Link>` component will refresh the entire page when the route jumps, but useRouter will not refresh the entire page. Is there any way to use `<Link>` without refreshing the entire page? Use the same layout between pages.

Someone can point me to the good direction? This problem has troubled me for a long time. I wonder if anyone has encountered a similar situation. Finally, thank you for taking the time to read my question, thank you!

Thanks!

**Comment by MarouaneBenbetka** [comment]
You can use <Link> without refreshing the entire page, ensuring you maintain the same layout across pages. This solution is in nextjs 

### app.jsx
```
import { useEffect } from 'react';
import { useRouter } from 'next/router';

function MyApp({ Component, pageProps }) {
  const router = useRouter();

  useEffect(() => {
    const handleRouteChange = (url) => {
      console.log('App is changing to: ', url)
    }

    router.events.on('routeChangeStart', handleRouteChange);

    return () => {
      router.events.off('routeChangeStart', handleRouteChange);
    }
  }, [router.events]);

  return <Component {...pageProps} />
}

export default MyApp;

```

### Navbar 
```
import Link from 'next/link';

function Navigation() {
  return (
    <nav>
      <Link href="/about"><a>About Us</a></Link>
      <Link href="/services"><a>Our Services</a></Link>
      <Link href="/contact"><a>Contact</a></Link>
    </nav>
  )
}

export default Navigation;
```

if this answers your questioin mark it as answered
